src/bin/hexyl.rs

```diff
@@ -94,45 +96,73 @@ fn run() -> Result<(), Box<dyn std::error::Error>> {
                 .value_name("STYLE")
                 .possible_values(&["unicode", "ascii", "none"])
                 .default_value("unicode")
-                .help("Whether to draw a border with Unicode characters, ASCII characters, or none at all"),
+                .help(
+                    "Whether to draw a border with Unicode characters, ASCII characters, \
+                    or none at all",
+                ),
         )
         .arg(
             Arg::with_name("display_offset")
                 .short("o")
                 .long("display-offset")
                 .takes_value(true)
                 .value_name("N")
-                .help("Add N bytes to the displayed file position. The N argument can also include \
-                a unit (see `--length` for details)"),
+                .help(
+                    "Add N bytes to the displayed file position. The N argument can also \
+                    include a unit (see `--length` for details)",
+                ),
         );
 
     let matches = app.get_matches_safe()?;
 
     let stdin = io::stdin();
 
-    let mut reader: Input = match matches.value_of("FILE") {
+    let mut reader = match matches.value_of("FILE") {
         Some(filename) => Input::File(File::open(filename)?),
         None => Input::Stdin(stdin.lock()),
     };
 
     let block_size = matches
         .value_of("block_size")
-        .and_then(|bs| bs.parse().ok().and_then(PositiveI64::new))
+        .map(|bs| {
+            bs.parse::<i64>()
+                .map_err(|e| anyhow!(e))
+                .and_then(|x| {
+                    PositiveI64::new(x)
+                        .ok_or_else(|| anyhow!("negative block sizes don't make sense"))
+                })
+                .context(anyhow!(
+                    "failed to parse `--block-size` arg {:?} as positive integer",
+                    bs
+                ))
+        })
+        .transpose()?
```

Comment:
Glad to see others using the `.map(…).transpose()?` pattern as well. `transpose` is such a useful helper. I'm happy it's finally in `std`.